core/base/src/TObject.cxx

```diff
@@ -779,36 +779,40 @@ void TObject::SavePrimitiveConstructor(std::ostream &out, TClass *cl, const char
    out << variable_name << " = new " << cl->GetName() << "(" << constructor_agrs << ");\n";
 }
 
+
 ////////////////////////////////////////////////////////////////////////////////
-/// Save array in the output stream "out".
+/// Save array in the output stream "out" as vector.
 /// Create unique variable name based on prefix value
+/// Returns name of vector which can be used in constructor or in other places of C++ code
 
-TString TObject::SavePrimitiveArray(std::ostream &out, const char *prefix, Int_t len, Double_t *arr, Bool_t empty_line)
+TString TObject::SavePrimitiveVector(std::ostream &out, const char *prefix, Int_t len, Double_t *arr, Bool_t empty_line)
 {
-   thread_local int arrid = 0;
+   thread_local int vectid = 0;
 
-   TString arrname = TString::Format("%s_darr%d", prefix, arrid++);
+   TString vectame = TString::Format("%s_vect%d", prefix, vectid++);
 
    if (empty_line)
       out << "   \n";
 
-   out << "   Double_t " << arrname << "[" << len << "] = { ";
+   out << "   std::vector<Double_t> " << vectame;
```

Comment:
why not std::array<Double_t, len> instead of std vector?

Reply:
Yes, std::array allocated in stack, which is much smaller than heap.
My doubt - if proposed syntax also has some side-effects. If interim array allocated in stack - one also will get same problems. I will try to test it.

Reply:
`TGraph` with 500K points works  - at least creation and storage of such object in macro.
Painting is other story.

Reply:
>  If interim array allocated in stack 

It should not, it should store directly in the object file data segment.  

Another alternative would have been to use a function static std::array (or regular array).